File: Python/for_if_while_try_def.py
# 1. For loop
fruits = ["apple", "banana", "orange"]
for fruit in fruits:
    print(fruit)

# 2. If statement
number = 10
if number > 0:
    print("Number is positive")
elif number == 0:
    print("Number is zero")
else:
    print("Number is negative")

# 3. While loop
counter = 0
while counter < 5:
    print("Counter:", counter)
    counter += 1

# 4. Try-except block
try:
    result = 10 / 0
except ZeroDivisionError:
    print("Cannot divide by zero")

# 5. Function definition
def greet(name):
    print("Hello,", name)

greet("Alice")

File: Go/for_if_while_try_def.go
package main

import "fmt"

func main() {
	// 1. For loop
	fruits := []string{"apple", "banana", "orange"}
	for _, fruit := range fruits {
		fmt.Println(fruit)
	}

	// 2. If statement
	number := 10
	if number > 0 {
		fmt.Println("Number is positive")
	} else if number == 0 {
		fmt.Println("Number is zero")
	} else {
		fmt.Println("Number is negative")
	}

	// 3. For loop with a condition
	counter := 0
	for counter < 5 {
		fmt.Println("Counter:", counter)
		counter++
	}

	// 4. Defer statement
	defer fmt.Println("End of program")

	// 5. Function definition
	greet("Alice")
}

func greet(name string) {
	fmt.Println("Hello,", name)
}
